Signal/src/ViewControllers/HomeView/HomeViewCell.m

```diff
@@ -178,34 +165,71 @@ - (void)configureWithThread:(TSThread *)thread
     [self updateAvatarView];
 
     self.payloadView.spacing = 0.f;
-    self.topRowView.spacing = ceil([HomeViewCell scaleValueWithDynamicType:5]);
+    self.topRowView.spacing = self.topRowHSpacing;
 
     // We update the fonts every time this cell is configured to ensure that
     // changes to the dynamic type settings are reflected.
     self.snippetLabel.font = [self snippetFont];
     self.snippetLabel.attributedText =
         [self attributedSnippetForThread:thread blockedPhoneNumberSet:blockedPhoneNumberSet];
 
-    self.dateTimeLabel.attributedText = [self attributedStringForDate:thread.lastMessageDate];
-    self.dateTimeLabel.textColor = hasUnreadMessages ? [UIColor ows_materialBlueColor] : [UIColor ows_darkGrayColor];
+    self.dateTimeLabel.text = [self stringForDate:thread.lastMessageDate];
+
+    if (hasUnreadMessages) {
+        self.dateTimeLabel.textColor = [UIColor ows_blackColor];
+        self.dateTimeLabel.font = self.dateTimeFont.ows_mediumWeight;
+    } else {
+        self.dateTimeLabel.textColor = [UIColor lightGrayColor];
+        self.dateTimeLabel.font = self.dateTimeFont;
+    }
 
     NSUInteger unreadCount = [[OWSMessageUtils sharedManager] unreadMessagesInThread:thread];
-    if (unreadCount > 0) {
-        [self.topRowView addArrangedSubview:self.unreadBadgeContainer];
+    if (unreadCount == 0) {
+        [self.viewConstraints addObject:[self.payloadView autoPinTrailingToSuperviewMargin]];
+    } else {
+        [self.contentView addSubview:self.unreadBadge];
 
-        self.unreadLabel.font = [UIFont ows_dynamicTypeCaption1Font];
-        self.unreadLabel.text = [OWSFormat formatInt:MIN(99, (int)unreadCount)];
+        self.unreadLabel.text = [OWSFormat formatInt:unreadCount];
```

Comment:
We need to update `unreadLabel.font` here or it won't reflect changes to dynamic type size.

Reply:
Good catch, thanks!